### ISSUE TITLE
Headsup: Memory leak with Powershell Core 7.3.0 - please stay with 7.2.7

### ISSUE THREAD

**Issue by RainbowMiner** [RainbowMiner/RainbowMiner#2211]
It seems like there is a severe memory leak in Powershell Core 7.3.0. I am investigating this, so that I can raise an issue at the Powershell Core github.

Until this is clarified, I strongly recommend you stay with Powershell Core 7.2.7.

If you have already installed 7.3.0, monitor your memory usage of pwsh/pwsh.exe closely and if it increases constantly, please consider to uninstall the 7.3.0 and then re-install the 7.2.7 version.

**Comment by UselessGuru** [comment]
There is also a memory leak problem with NemosMiner (any version of PS show simily symptoms :-( ).
So far I could not  figure out what exactly causes the memory leaks.
The only change that improved the issue was replacing some 'ConvertTo-Json | ConverFrom-Json' with $Object.Clone().

**Comment by AnthonyBe** [comment]
I see what you mean!

```
[2022-11-20 00:00:54] INFO: Memory usage: 1,292.6 MB
...
[2022-11-20 21:34:43] INFO: Memory usage: 1,858.5 MB
```

**Comment by RainbowMiner** [comment]
Yes, so better uninstall 7.3.0 and then re-install the 7.2.7 - there is no memory leak with that one.

**Comment by RainbowMiner** [comment]
Ok, one memory leak problem with Powershell Core 7.3.0 has been confirmed: https://github.com/PowerShell/PowerShell/issues/18627

Since we make heavy use of Start-Job / Start-ThreadJob it explains the memory leak. Even Receive-Job doesn't help here. So I reiterate the recommendation: stick to Powershell Core 7.2.7, do NOT use Powershell Core 7.3.0

**Comment by AnthonyBe** [comment]
I'm guessing, based on the tag changes on the Git issue after you asked yesterday, that this issue isn't resolved in PowerShell 7.3.1?

**Comment by RainbowMiner** [comment]
Yes, it pretty much looks like that. I'll add the "solved" tag to our thread as soon, as there is a Powershell Core version, that has solved the issue.
